virttest/qemu_devices/qdevices.py

```diff
@@ -907,23 +907,56 @@ def hotplug_qmp(self):
         if params.get("host-nodes"):
             host_nodes = list(map(int, params["host-nodes"].split()))
             params["host-nodes"] = host_nodes
+        for k in params:
+            if params[k] == "yes":
+                params[k] = True
+            if params[k] == "no":
+                params[k] = False
         kwargs = {"qom-type": backend,
                   "id": params.pop("id"),
                   "props": dict(params)}
         return "object-add", kwargs
 
     def verify_unplug(self, out, monitor):
-        out = monitor.info("memory-devices", debug=False)
-        memdev = "/object/%s" % self.get_qid()
-        if memdev not in out:
-            return True
-        return False
+        """
+        :param out: Output of the unplug command
+        :param monitor: Monitor used for unplug
+        :return: True when successful, False when unsuccessful
+        """
+        out = monitor.info("memdev", debug=False)
+        memdev = self.get_qid()
+        for dev in out:
+            if dev["id"] == memdev:
+                return False
+        return True
 
     def verify_hotplug(self, out, monitor):
-        out = monitor.info("memory-devices", debug=False)
-        memdev = "/object/%s" % self.get_qid()
-        if memdev in out:
-            return True
+        """
+        :param out: Output of the hotplug command
+        :param monitor: Monitor used for hotplug
+        :return: True when successful, False when unsuccessful
+        """
+        out = monitor.info("memdev", debug=False)
+        memdev = self.get_qid()
+        params = self.params.copy()
+        for dev in out:
+            if dev["id"] == memdev:
+                if not params.get("host-nodes"):
+                    return False
+                host_nodes = list(map(int, params["host-nodes"].split()))
+                args = (params["size"], "B", 1024)
+                size = int(float(utils_misc.normalize_data_size(*args)))
+                if dev["size"] != size or dev["host-nodes"] != host_nodes:
+                    return False
+                dev.pop("size")
+                dev.pop("host-nodes")
+                for k in dev:
+                    if params.get(k):
+                        params[k] = True if params[k] == "yes" else params[k]
+                        params[k] = False if params[k] == "no" else params[k]
```

Comment:
Duplicate with L910~L914 ?

Reply:
@yanan-fu Not exactly, except `if params.get(k):`.   I'm not sure if it is necessary to create a function for it.

Reply:
@huangyum  From my understanding, it is same, no need to handle it twice, thx

Reply:
Agree with @yanan-fu You don't need #line 955-956

Reply:
@huangyum or, why not use  True, False in case cfg directly.

Reply:
@yanan-fu Because qemu cmd and qmp cmd accept different values for some options.  e.g. For option "dump",  qemu cmd accepts "on/yes/y/off/no/n" while qmp cmd only accepts "True/False".

Reply:
@yanan-fu @vivianQizhu  How about a function which changes a value from "yes" to "True" and "no" to "False"?  That's the only common part.